content/vendor/guides/configuring-application-snapshots.md

```diff
@@ -0,0 +1,345 @@
+  Per the Velero documentation, "The hook is considered in error if the command exceeds the timeout." so make sure this value exceeds how long you expect the command to take.
+- Add a new volume to the pod and mount it to the new folder containing the ```pg_dump``` output.
+
+Below is the same definition file with the modifications:
+
+```shell
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: app-direct-postgresql
+  labels:
+    app: postgresql
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgresql
+  serviceName: postgres-service
+  template:
+    metadata:
+      labels:
+        app: postgresql
+      annotations: 
+        pre.hook.backup.velero.io/command: '["/bin/bash", "-c", "PGPASSWORD=$POSTGRES_PASSWORD pg_dump -U $POSTGRES_USER -d appdirectdb -h 127.0.0.1 > /backup/backup.sql"]'
+        pre.hook.backup.velero.io/timeout: 3m
+        backup.velero.io/backup-volumes: postgresql-backup 
+    spec:
+      containers:
+      - name: app-direct-postgresql
+        image: "postgres:9.6"
+        env:
+        - name: POSTGRES_USER
+          value: "postgres"
+          # Required for pg_isready in the health probes....
+        - name: PGUSER
+          value: "postgres"
+        - name: POSTGRES_DB
+          value: "postgres"
+        - name: POSTGRES_INITDB_ARGS
+          value: ""
+        - name: PGDATA
+          value: /var/lib/postgresql/data/pgdata
+        - name: POSTGRES_PASSWORD
+          value: "postgres"
+        - name: POD_IP
+          valueFrom: { fieldRef: { fieldPath: status.podIP } }
+        ports:
+        - name: postgresql
+          containerPort: 5432
+        livenessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - exec pg_isready --host $POD_IP
+          initialDelaySeconds: 60
+          timeoutSeconds: 5
+          failureThreshold: 6
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - exec pg_isready --host $POD_IP
+          initialDelaySeconds: 5
+          timeoutSeconds: 3
+          periodSeconds: 5
+        resources:
+          requests:
+            cpu: 100m
+            memory: 256Mi
+
+        volumeMounts:
+        - name: postgresql-vct
+          mountPath: /var/lib/postgresql/data/pgdata
+          subPath: postgresql-db
+        - name: postgresql-backup
+          mountPath: /backup
+          subPath: postgresql-db
+  volumeClaimTemplates:
+  - metadata:
+      name: postgresql-vct
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 2Gi
+  - metadata:
+      name: postgresql-backup
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 2Gi
+
+```
+{{< notes title="Note" >}}
+Note that in the command in the label annotation we are backing up the appdirectdb database, which does not exist when the application is first deployed. 
```

Comment:
I don't really grok this -- I'm hoping that later, the tutorial will have me create the DB in appDirect before running a backup, but if not, again, we should tell the user exactly what to do to do an E2E snapshot workflow, not give them a bunch of options.